app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java

```diff
@@ -87,4 +72,57 @@ public void onDisplayNetworkError(boolean networkNotAvailable) {
     public void onRequestFinish(@NotNull UiFinishReason uiFinishReason) {
         finish();
     }
+
+    /**
+     * Required Override when using [WebUiNetworkErrorFragment.Listener]
+     * <p>
+     * This is the returning call from the [WebUiNetworkErrorFragment] when the ok/close button is
+     * pressed. If using a fragment integration and not subclassing [WebUiActivity], then the
+     * error view will not know how to handle the event.
+     */
+    @Override
+    public void onNetworkErrorClose() {
+        onRequestFinish(UiFinishReason.UNABLE_TO_LOAD_SPA);
+    }
+
+    // -----WebUiNetworkErrorFragment.Listener Overrides -----
+
+    /**
+     * Required Override when using [WebUiNetworkErrorFragment.Listener]
+     * <p>
+     * This is the returning call from the [WebUiNetworkErrorFragment] when the retry button is
+     * pressed. If using a fragment integration and not subclassing [WebUiActivity], then the
+     * error view will not know how to handle the event.
+     */
+    @Override
+    public void onNetworkErrorRetry() {
+        loadWebUi();
+    }
+
+    private void loadWebUi() {
+        if (this.isFinishing()) {
+            return;
+        }
+
+        WebUiFragment fragment = new WebUiFragment();
```

Comment:
I do think for both this and the kotlin version, we should update the fragment creation.
Right now, for either the error fragment or the web fragment we'll never reuse an existing instance, we always create a new fragment every time.
We should use a `supportFragmentManager.getByTag() ?: new Fragment()` method so we can reuse any existing fragments.

Especially now that we're explicitly switching between fragments (potentially multiple times) here

Reply:
Kotlin side was already reusing the web fragment, but I also updated to reuse the error fragment. Also applied the same logic to the java version